pulp_docker/app/authorization.py

```diff
@@ -0,0 +1,172 @@
+import base64
+import hashlib
+import random
+import uuid
+
+import jwt
+
+from datetime import datetime
+from aiohttp import web
+
+from django.conf import settings
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+
+from pulpcore.plugin.content import Handler
+
+TOKEN_EXPIRATION_TIME = 300
+
+KNOWN_SERVICES = ['v2']
+ANONYMOUS_USER = ''
+
+
+class AuthorizationService(Handler):
+    """
+    A class responsible for generating and managing a Bearer token.
+
+    This class represents a token server which manages and grants permissions
+    according to a user's scope.
+    """
+
+    async def get_token(self, request):
```

Comment:
i suggest to rename it to generate_token, we already have _get_token and it can be a bit confusing